### PR TITLE
[1.5] Fix registry_auth logic for upgrades

### PR DESCRIPTION
**Pull request by michaelgugino**
Currently, the logic for registry authentication is
not implemented correctly to account for upgrades of
containerized hosts.

Additionally, the logic to account for multiple runs
of openshift-ansible might cause registry authentication
credentials to not be mounted inside of containerized hosts.

This commit adds the necessary logic to ensure containerized
hosts retain registry credentials.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1494470
(cherry picked from commit 8d17936a6fa012a4296bbb8ee9c72dd494e2e94d)

Backports: https://github.com/openshift/openshift-ansible/pull/5502

### CODE REVIEW

File: roles/openshift_node_upgrade/tasks/registry_auth.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Check for credentials file for registry auth
+  stat:
+    path: "{{ oreg_auth_credentials_path }}"
```
Comment (tbielawa) [inline roles/openshift_node_upgrade/tasks/registry_auth.yml]: So this file is now duplicated across the `openshift_node` and `openshift_node_upgrade` roles. Not very DRY. Any way around that?

Comment (michaelgugino) [reply roles/openshift_node_upgrade/tasks/registry_auth.yml]: Next cycle I would like to see openshift_node and openshift_node_upgrade roles combined into one.  We could use include_role.  That will remove lots of duplicated items and work-arounds (like the set-facts I have to do in openshift_master).  Same applies to openshift_master.

**Pull request merged**